include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -108,7 +116,9 @@ struct envelope_linear_on_spheroid
             Box segment_mbr;
             dispatch::envelope
                 <
-                    typename std::iterator_traits<iterator_type>::value_type
+                    typename std::iterator_traits<iterator_type>::value_type,
+                    0,
+                    DimensionCount
```

Comment:
Actually, maybe not? If here all of the coordinates were handled then all of the coordinates of the resulting envelope could be updated here, inside this loop. Then it wouldn't be needed to call the `dispatch::envelope` for the dimensions > 2 and therefore have 2 loops instead of 1. Or am I missing something?